sljit_src/sljitNativeX86_common.c

```diff
@@ -3117,15 +3121,19 @@ SLJIT_API_FUNC_ATTRIBUTE struct sljit_put_label* sljit_emit_put_label(struct slj
 SLJIT_API_FUNC_ATTRIBUTE void sljit_set_jump_addr(sljit_uw addr, sljit_uw new_target, sljit_sw executable_offset)
 {
 	SLJIT_UNUSED_ARG(executable_offset);
+	SLJIT_UPDATE_WX_FLAGS((void*)addr, (void*)addr + (sizeof(sljit_uw)), 0);
 #if (defined SLJIT_CONFIG_X86_32 && SLJIT_CONFIG_X86_32)
 	sljit_unaligned_store_sw((void*)addr, new_target - (addr + 4) - (sljit_uw)executable_offset);
 #else
 	sljit_unaligned_store_sw((void*)addr, (sljit_sw) new_target);
 #endif
+	SLJIT_UPDATE_WX_FLAGS((void*)addr, (void*)addr + (sizeof(sljit_uw)), 1);
 }
 
 SLJIT_API_FUNC_ATTRIBUTE void sljit_set_const(sljit_uw addr, sljit_sw new_constant, sljit_sw executable_offset)
 {
 	SLJIT_UNUSED_ARG(executable_offset);
+	SLJIT_UPDATE_WX_FLAGS((void*)addr, (void*)addr + (sizeof(sljit_sw)), 0);
 	sljit_unaligned_store_sw((void*)addr, new_constant);
```

Comment:
alternatively why not do the SLJIT_UPDATE_WX_FLAGS inside sljit_unaligned_store_sw to cover both sljit_set_jump_{addr, const}()

Reply:
this needs the same casting change done above to avoid (void *) arithmetic :

`
(void*)(addr + (sizeof(sljit_sw))
`